### PR TITLE
DOC: gallery: join segmentation: enhancement

### PR DESCRIPTION
**Pull request by sciunto**
## Description

I made few changes in an example:

* use skimage functions instead of ndi.
* add a comment in the code
* remove a useless img_as_float
* and few other stuffs.

### CODE REVIEW

File: doc/examples/segmentation/plot_join_segmentations.py [hunk 1]
```diff
@@ -11,26 +11,29 @@
...
 from skimage.segmentation import slic, join_segmentations
 from skimage.morphology import watershed
 from skimage.color import label2rgb
-from skimage import data, img_as_float
+from skimage import data
 
-coins = img_as_float(data.coins())
+coins = data.coins()
 
 # make segmentation using edge-detection and watershed
```
Comment (jni) [inline doc/examples/segmentation/plot_join_segmentations.py]: "Make [...] watershed." (proper sentence)

File: doc/examples/segmentation/plot_join_segmentations.py [hunk 2]
```diff
@@ -11,26 +11,29 @@
...
 edges = sobel(coins)
+
+# Identifies some background and foreground pixels from the intensity values.
```
Comment (jni) [inline doc/examples/segmentation/plot_join_segmentations.py]: Identify

**Comment by sciunto** [comment]
Thanks @jni I fixed that.

**Comment by blink1073** [comment]
@sciunto, can you please try a rebase to see if that clears up the appveyor failure from #2497?

**Comment by sciunto** [comment]
Done.

**Comment by blink1073** [comment]
That did it, sorry for the disruption ;).

**Pull request merged**